object_detection/Mask_RCNN/mask_rcnn_run_loop.py

```diff
@@ -0,0 +1,181 @@
+# flags.DEFINE_string('checkpoint_dir', '',
+#                    'Directory containing checkpoints to evaluate, typically '
+#                    'set to `train_dir` used in the training job.')
+flags.DEFINE_string('eval_dir', '',
+                    'Directory to write eval summaries to.')
+
+flags.DEFINE_boolean('run_once', False, 'Option to only run a single pass of '
+                                        'evaluation. Overrides the `max_evals`'
+                                        ' parameter in the provided config.')
+flags.DEFINE_float('box_min_ap', 1, 'Option to run until the box average'
+                                    'precision reaches this number')
+flags.DEFINE_float('mask_min_ap', 1, 'Option to run until the mask average'
+                                     'precision reaches this number')
+FLAGS = flags.FLAGS
+
+
+def train(_):
+  # TODO(mehdi): training 1 epoch
+  pass
+
+
+def evaluate(_):
+  # TODO(mehdi): evaluating with mask and box metrics
+  pass
+
+
+def main(_):
+  assert FLAGS.train_dir, '`train_dir` is missing.'
+  assert FLAGS.pipeline_config_path, '`pipeline_config_path` is missing'
+  assert FLAGS.checkpoint_dir, '`checkpoint_dir` is missing.'
+  assert FLAGS.eval_dir, '`eval_dir` is missing.'
+
+  configs = config_util.get_configs_from_pipeline_file(
+      FLAGS.pipeline_config_path)
+  if FLAGS.task == 0:
+    tf.gfile.MakeDirs(FLAGS.train_dir)
+    tf.gfile.Copy(FLAGS.pipeline_config_path,
+                  os.path.join(FLAGS.train_dir, 'pipeline.config'),
+                  overwrite=True)
+
+  tf.gfile.MakeDirs(FLAGS.eval_dir)
+  tf.gfile.Copy(FLAGS.pipeline_config_path,
+                os.path.join(FLAGS.eval_dir, 'pipeline.config'),
+                overwrite=True)
+
+  model_config = configs['model']
+
+  train_config = configs['train_config']
+  train_input_config = configs['train_input_config']
+
+  eval_config = configs['eval_config']
+  if FLAGS.eval_training_data:
+    eval_input_config = configs['train_input_config']
+  else:
+    eval_input_config = configs['eval_input_config']
+
+  train_model_fn = functools.partial(model_builder.build,
+                                     model_config=model_config,
+                                     is_training=True)
+  eval_model_fn = functools.partial(model_builder.build,
+                                    model_config=model_config,
+                                    is_training=False)
+
+  def get_next(config):
+    return dataset_util.make_initializable_iterator(
+        dataset_builder.build(config)).get_next()
+
+  train_input_dict_fn = functools.partial(get_next, train_input_config)
+  eval_input_dict_fn = functools.partial(get_next, eval_input_config)
+
+  env = json.loads(os.environ.get('TF_CONFIG', '{}'))
+  cluster_data = env.get('cluster', None)
+  cluster = tf.train.ClusterSpec(cluster_data) if cluster_data else None
+  task_data = env.get('task', None) or {'type': 'master', 'index': 0}
```

Comment:
Might be cleaner to do `task_data =  env.get('task', {'type': 'master', 'index': 0})`